planner/core/exhaust_physical_plans.go

```diff
@@ -802,13 +802,14 @@ func (p *LogicalJoin) buildIndexJoinInner2TableScan(
 		if !pkMatched {
 			return nil
 		}
-		innerTask = p.constructInnerTableScanTask(ds, pkCol, outerJoinKeys, us, false, false, avgInnerRowCnt)
+		ranges := ranger.FullIntRange(mysql.HasUnsignedFlag(pkCol.RetType.Flag))
```

Comment:
We change `ranges` here. Besides calculating seek cost of probe-side, it also affects the calculation of other costs since changing of `ranges` causes changing of `rowCount`. I'm not sure that whether `helper.chosenRanges.Range()` is suitable for estimating row count since it comes from `(*indexJoinBuildHelper).buildTemplateRange`, which seems to fill empty datum in some ranges.

Reply:
After talking with @qw4990, `ranges` here only affects calculating seek cost of probe-side, it doesn't affect calculating row count or other stats. Hence it's OK.

Reply:
`ranges` won't affect `rowCount` here ~
And actually, `helper.chosenRanges.Range()` is the only range we can use to calculate seek cost. In most cases, `chosenRanges` is `[NULL, NULL]` and `len(chosenRanges)` is 1, which is equal to what we use now `len(FullRange)`. For queries like `select /*+ use_index(t1, ab) */ * from t1, t2 where t1.a in (1, 2, 3) and t1.b=t2.b`, it can be large than 1 and more accurate than `len(FullRange)`.